modules/dpu-enable-dpu-hw-offloading.adoc

```diff
@@ -0,0 +1,341 @@
+kind: SriovNetworkPoolConfig
+metadata:
+  name: default
+  namespace: openshift-sriov-network-operator
+spec:
+  ovsHardwareOffloadConfig:
+    name: dpu-host <1>
+----
+<1> The name here is the same as the machine config pool (MCP) name created in step 1.
+.. Apply the configuration:
++
+[source,terminal]
+----
+$ oc create -f sriov-pool-config.yaml.yaml
+----
+
+. Create a SriovNetworkNodePolicy to configure the virtual functions (VFs) on the hosts.
+
+.. Save the YAML in the `SriovNetworkNodePolicy.yaml` file:
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: policy-mlnx-bf <1>
+  namespace: openshift-sriov-network-operator
+spec:
+  resourceName: mlnx_bf <2>
+  nodeSelector:
+    node-role.kubernetes.io/dpu-host: "" <3>
+  priority: 99 <4>
+  numVfs: 4 <5>
+  nicSelector: <6>
+    vendor: "15b3" <7>
+    deviceId: "a2d6" <8>
+    pfNames: ['ens1f0'] <9>
+    rootDevices: ['0000:3b:00.0'] <10>
+----
++
+<1> The name for the custom resource object.
++
+<2> The resource name of the SR-IOV network device plug-in. You can create multiple SR-IOV network node policies for a resource name.
++
+<3> The node selector specifies the nodes to configure. Ensure this is consistent with the `nodeSelector` of the MCP created in step 1.
++
+<4> Optional: The priority is an integer value between 0 and 99. A smaller value receives higher priority. For example, a priority of 10 is a higher priority than 99. The default value is 99.
++
+<5> The number of the virtual functions (VF) to create for the SR-IOV physical network device. For a Mellanox NIC, the number of VFs cannot be larger than 128.
++
+<6> The NIC selector identifies the device for the Operator to configure. You do not have to specify values for all the parameters. It is recommended to identify the network device with enough precision to avoid selecting a device unintentionally.
++
+<7> The vendor hexadecimal code of the SR-IOV network device. Vendor id `15b3` is for Mellanox devices.
++
+<8> The device hexadecimal code of the SR-IOV network device. For example, `a2d6` is the device ID for a Bluefield-2 DPU device.
++
+<9> An array of one or more physical function (PF) names for the device.
++
+<10> An array of one or more PCI bus addresses for the PF of the device. Provide the address in the following format: `0000:02:00.1`.
++
+.. Create the SriovNetworkNodePolicy object:
++
+[source,terminal]
+----
+$ oc create -f SriovNetworkNodePolicy.yaml
+----
+
+. Create a Cluster Network Operator (CNO) ConfigMap in the tenant cluster setting the mode to `dpu-host`.
+
+.. Save the YAML in the `sriovdpuconfigmap.yaml` file:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: dpu-mode-config
+    namespace: openshift-network-operator
+data:
+    mode: "dpu-host"
+immutable: true
+----
+
+.. Run the following command:
++
+[source,terminal]
+----
+$ oc create -f sriovdpuconfigmap.yaml
+----
+
```

Comment:
We need to disable OVS on the hosts. As we don't need OVS anymore on the host side when ovnkube is running in the dpu-host mode.

It can be done by applying a MachineConfig to the dpu-host pool
```
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: dpu-host
  name: disable-ovs
spec:
  config:
    ignition:
      version: 3.1.0
    systemd:
      units:
      - mask: true
        name: ovs-vswitchd.service
      - enabled: false
        name: ovs-configuration.service
```

Reply:
Hi @pliurh I updated please see section here: http://file.emea.redhat.com/kquinn/TELCODOCS-472/hardware_enablement/dpu-hardware-offload.html#configuring-support-for-hardware-offloading-in-the-tenant-cluster